modules/5_install/install.tf

```
@@ -400,7 +401,7 @@ resource "null_resource" "powervs_config" {
 
 resource "null_resource" "upgrade" {
   depends_on = [null_resource.install, null_resource.powervs_config]
-  count      = var.upgrade_version != "" ? 1 : 0
+  count      = (var.upgrade_version != "" || var.upgrade_image != "") != "" ? 1 : 0
```

@varad-ahirwadkar Did you test all the combinations here? As per my knowledge, this will always evaluate as `true` (hence count as 1).

---

I tested with three scenarios:
By using blank values -> successfully deployed cluster
By setting both variables to check precedence -> successfully deployed cluster and upgraded to `upgrade_image`
By setting only upgrade_image ->  successfully deployed cluster and upgraded to `upgrade_image`

---

See this from Terraform console:
```
> ("" != "" || "" != "")
false
> ("" != "" || "" != "") != ""
true
```

You are comparing `false != ""` which is always `true`.

---

> By using blank values -> successfully deployed cluster

Deployment may be successful but we don't need to run the upgrade resource.

```suggestion
  count      = var.upgrade_version != "" || var.upgrade_image != "" ? 1 : 0
```

---

Hi @yussufsh, I have made this change and re-tested the above three scenarios.
Thanks.